### ISSUE TITLE
[Bug]: Cheapest Station Boolean true even if not cheapest

### ISSUE THREAD

**Issue by Diginix** [iobroker-community-adapters/ioBroker.tankerkoenig#111]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

3 of 4 stations has price 1.699 and one has 1.719. All 4 stations has cheapest = true

### Adapter version

3.3.2

### Environment

```markdown
- Node: 16.19.9
- npm: 8.19.3
- JS-Controller: 4.0.24
- Operating system: Ubuntu LTS
```

### Screenshots & Log

![image](https://user-images.githubusercontent.com/39678667/210231870-ff99bdb4-55e2-4918-b2de-88d86de2b7e4.png)

**Comment by xXBJXx** [comment]
Check if `tankerkoenig.0.stations.cheapest.e5.cheapest_stations` contains the station as well

**Comment by Diginix** [comment]
No this JSON contains only the 3 correct ones.

**Comment by xXBJXx** [comment]
Is the station closed or open?

**Comment by Diginix** [comment]
open

**Comment by xXBJXx** [comment]
ok i will look at it again

**Comment by Diginix** [comment]
All 4 are in the same city and has the same "conditions" atm. Only the boolean flag seams not to be toggled if another one becomes cheaper.

**Comment by xXBJXx** [comment]
I cannot reproduce the error, I have now created 7 stations and run the times a few hours, let the flag is set as well as the names in the `tankerkoenig.0.stations.cheapest.e5.cheapest_stations`.
Was the state about have updated ⇨
![image](https://user-images.githubusercontent.com/32600934/210251136-f1d7a054-d034-4b59-b83e-0107f058df71.png)

**Comment by Diginix** [comment]
Also the update time was the same as for all other stations.
At the moment the real cheapest station is flaged, all others not.
I have 8 stations, 4 for 2 different cities. But this doesn't matter for the logic. Only the cheapest price over all should be flagged. No matter how much stations. I will have a look if it happens again. But I don't know what more I could provide for finding the cause.

**Comment by xXBJXx** [comment]
> btw: I use E10, not E5
Maybe you should have said that earlier because there is a bug because not the e10, but the e5 are displayed, I have had a written error in the variable.
That could be the problem.

**Comment by xXBJXx** [comment]
I fix it in version 3.3.3

**Comment by Diginix** [comment]
Sorry, doesn't thought that the fuel type could matter. But good to hear that you found something related.

**Comment by smartcuc** [comment]
Cool. Having the same issue und good to hear, that there wi be fix.

**Comment by xXBJXx** [comment]
Released in v3.3.3
